lib/mocha.js

```diff
@@ -180,6 +180,17 @@ Mocha.prototype.reporter = function (reporter, reporterOptions) {
 };
 
 /**
+ * Set reporter options from mocha.setup call
+ * @param {Object} reporterOptions optional options
+ */
+Mocha.prototype.reporterOptions = function (reporterOptions) {
+
+  this.options.reporterOptions = this.options.reporterOptions || reporterOptions;
```

Comment:
This code as written will ignore the specified options if `this.options.reporterOptions` already exists; that behavior is likely to be confusing at best and may even be erroneous.

There's some question, in my mind, as to whether existing options should be *merged* with new options, e.g. using `Object.assign(this.options.reporterOptions, reporterOptions)`, or just overwritten altogether. Assuming this function is desirable, whether to merge or overwrite would merit further discussion.

Reply:
I'm totally agreed ! I don't known what will be the options management inside the mocha core, and what are the desired usage for it.

All i known for the moment is that the current Mocha prototypes doesn't allow to use the setup methods for all available options in Mocha.

Reply:
I will add Object.assign in next commit. The only case where this could happen is, i create a new Mocha instance with reporterOptions as arguments, and next call setup methods on it. Is this a real use case ? Anyway this need to be properly handle... so...